src/10-ipfs-0-4-0-released/index.md

```diff
@@ -0,0 +1,42 @@
+---
+date: 2016-01-10
+id: 10-ipfs-0-4-0-released
+template: tmpl/layouts/post.html
+baseurl: ..
+breadcrumbs:
+  - {name: "10-ipfs-0-4-0-released", link: "./" }
+tags: modules
+title: IPFS 0.4.0 has been released
+author: Kyle Drake
+collection: posts
+---
+
+[IPFS 0.4.0](https://github.com/ipfs/go-ipfs/releases/tag/v0.4.0) has been released! Among the many changes are a revamped implementation of the IPFS communication protocols, increased performance, improvements to IPNS, many bugfixes, and a couple new features to make IPFS even more powerful.
+
+### On Breaking Changes
+
+This is a **breaking change** release, with a major refactor and upgrade to the IPFS networking protocol. Because of the refactor, **all IPFS daemons earlier than 0.4.0 will not be able to communicate with the newest version**. It is strongly recommended that everyone running an IPFS node upgrades to the latest version as soon as possible, as these nodes will, after a certain time, no longer be able to communicate with the majority of the network until they are upgraded.
+
+Refactoring the protocol is not something to be done lightly. But at this early stage, this is necessary to ensure we have the right design for IPFS in place for the future. It's better to improve the protocol now during this alpha stage of the project than when there are a lot more people running nodes across a lot of different implementations.
+
+One of the important changes that's been made to the protocol means that *there should never be a breaking change like this again*. This is due to a change to allow nodes to announce the version of the protocol they are using when connecting to other IPFS nodes. The goal is to roll any future protocol changes into the implementations gradually, so that we can still support legacy protocols for a period of time, making it easier to deprecate old versions over time.
+
+## Why we're changing the protocol
+
+The most important change was to allow IPFS implementations to use pluggable stream multiplexers, such as [yamux](https://github.com/hashicorp/yamux), [spdystream](https://github.com/docker/spdystream), or [muxado](https://github.com/inconshreveable/muxado). Instead of locking IPFS permanently into a single multiplexer that won't work for every language or situation, this change allows the implementations to implement the multiplexers of their choosing.
```

Comment:
"... and then nodes will negotiate which multiplexer to use between them during the initial connection handshake"